models/AbstractModel.php

```diff
@@ -46,6 +53,15 @@ protected function getDataMapper():AbstractDataMapper {
             return $this->relatedMapper;
         }
 
+        /**
+         * Returns the Database connection
+         *
+         * @return mysqli
+         */
+        protected function connectToDB():mysqli {
+            return DBConnectionProvider::getConnection(IDBConnection::class);
+        }
+
```

Comment:
A new method will reduce the amount of repeated code and help with the testing process.